src/components/FullDetailView/FullDetailPage.js

```diff
@@ -57,71 +67,118 @@ export class FullDetailPage extends Component {
   doPromptContribute() {
-    const { component } = this.props
-    if (!Contribution.hasChange(component)) return
+    const { changes } = this.state
+    if (isEmpty(changes)) return
     this.refs.contributeModal.open()
   }
 
-  handleSave(changes) {
-    const { onSave, component } = this.props
+  handleSave() {
+    const { onSave, component, uiCurateResetDefinitionPreview } = this.props
+    const { changes } = this.state
     const newComponent = { ...component, changes }
-    onSave && onSave(component, newComponent)
+    this.setState({ changes: {} }, () => {
+      uiCurateResetDefinitionPreview()
+      onSave && onSave(component, newComponent)
+    })
   }
 
   handleClose() {
     const { onClose } = this.props
-    onClose()
-    this.setState({ visible: false })
-  }
-
-  // Function called when a data has been changed
-  onChange(item, value) {
-    const { component } = this.props
     const { changes } = this.state
-    this.setState({ changes: Contribution.onChange(component, changes, item, value) }, () =>
-      console.log(this.state.changes)
+    if (isEmpty(changes)) return onClose()
+    const key = `open${Date.now()}`
+    const btn = (
+      <Fragment>
+        <AntdButton
+          type="primary"
+          size="small"
+          onClick={() => {
+            this.close()
+            notification.close(key)
+          }}
+        >
+          Confirm
+        </AntdButton>
+        <AntdButton type="secondary" size="small" onClick={() => notification.close(key)}>
+          Dismiss Notification
+        </AntdButton>
+      </Fragment>
     )
+    notification.open({
+      message: 'Unsaved Changes',
+      description:
+        'Some information have been changed and are currently unsaved. Are you sure to continue without saving?',
+      btn,
+      key,
+      onClose: notification.close(key),
+      duration: 0
+    })
   }
 
-  getValue(field) {
-    const { component } = this.props
-    const { changes } = this.state
-    return Contribution.getValue(component, changes, field)
+  close() {
+    const { uiCurateResetDefinitionPreview, onClose } = this.props
+    this.setState({ changes: {} }, () => {
+      uiCurateResetDefinitionPreview()
+      onClose()
+    })
   }
 
-  classIfDifferent(field, className) {
+  // Function called when a data has been changed
+  onChange(item, value, type, transform) {
     const { component } = this.props
     const { changes } = this.state
-    return Contribution.classIfDifferent(component, changes, field, className)
+    this.setState({ changes: Contribution.applyChanges(component, changes, item, value, type, transform) }, () =>
+      this.previewDefinition()
+    )
   }
 
   render() {
-    const { path, component, definition, curation, harvest, modalView, visible } = this.props
+    const { path, definition, curation, harvest, modalView, visible, previewDefinition } = this.props
     const { changes } = this.state
-
     return modalView ? (
```

Comment:
suggest not using the ternary operator where the clauses are multiple lines as a "general" principle. Here you could just as well 

```javascript
if (modalView) return <Modal ...>
</Modal>
return <Grid ...>
</Grid>
```